design/vela-core/resourcetracker_design.md

```diff
@@ -0,0 +1,56 @@
+# ResourceTracker: Managing Resources behind Application
+
+- Owner: Da Yin (@somefive)
+- Reviewer: Jian Li (@leejanee), Jianbo Sun (@wonderflow)
+- Date: 12/10/2021
+- Status: Implemented
+
+## Intro
+
+As the release of Workflow in KubeVela v1.1, resources dispatched by application can be extremely dynamic.
+Instead of simply declaring in Application Component, users can have advanced control for applied resources by leveraging WorkflowSteps such as `ApplyObject` or operators such as `op.Delete`.
+Meanwhile, with EnvBinding Policy also released in KubeVela 1.1, users can have multiple same components but deployed in different clusters.
+These techniques raise new challenges for tracking and maintaining of resources.
+
+## Goal
+
+To tackle these challenges, a new architecture of ResourceTracker is proposed and implemented.
+Generally, there are several major technical changes compared to the previous version:
+
+1. Resources do not use OwnerReference to track their ResourceTracker anymore. In other word, the previous bi-directional binding is simplified into uni-directional, which allows us to have more flexible resource management strategies (such as releasing the control of resources).
+2. Resources rendered manifests are recorded in ResourceTracker optionally. Based on that, we can prevent configuration drift by leveraging the reconciling mechanism of the Kubernetes operator pattern. 
+3. ResourceTracker deletion now use finalizer and leverage ApplicationController to reconcile. This ensures the deletion of Application truly removes all managed resources. Also, it allows users to manage versioned resource manually. 
```

Comment:
Will a Terraform typed Application wait for Configuration and its auxiliaries objects to delete when it is deleting?

Reply:
When will a ResourceTracker be deleted?

Reply:
ResourceTracker will be deleted if all its managed resources have been deleted or controlled by other newer RTs. Therefore, a Terraform typed Application will hang in deleting status until the Configuration object is removed. For other auxiliaries, it created by Application, it will wait; if created by TerraformController, it will not.

Reply:
That's nice. It's what I was expecting:)